service/api/src/main/java/org/eclipse/kapua/KapuaDeviceMenagementException.java

```diff
@@ -0,0 +1,22 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Eurotech and/or its affiliates and others
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Eurotech - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.kapua;
+
+public class KapuaDeviceMenagementException extends KapuaException{
```

Comment:
Why are you creating a new exception here? `DeviceManagementException` didn't work for you?

Reply:
It didn't look like a good idea to handle `DeviceManagementException`  with the `KapuaExceptionHandler`, as this is an exception from kapua-device-commons package, so I created that new exception. Should I change this?
It seems like there is no unified way of handling errors in Kapua, some are shown directly to the user from service side and internal packages and for others the KapuaExceptionHandler is used. 
So, what approach should be used here?

Reply:
Well... `DeviceManagementException` is a `KapuaException` anyway, so from a strict inheritance perspective your new exception doesn't add any different semantics. On the other hand, if we don't have a dependency from `kapua-device-commons` already, it may be a good idea to not create one. Pinging @Coduz for his feedback!

Reply:
I'm fine with a more specific exception for the `DeviceManagementException`s.

I would just drop the `Kapua` and enforce that a specific set of error codes (as Enums)  is used.

```
public class DeviceMenagementException extends KapuaException{
[...]

public DeviceMenagementException(DeviceMenagementErrorCodes code) {
        super(code);
}

```

What you guys think?